packages/validator/src/services/indices.ts

```diff
@@ -71,7 +84,42 @@ export class IndicesService {
         newIndices.push(validatorState.index);
       }
     }
-
     return newIndices;
+  };
+}
+
+type Batch = string[][];
+const PUBKEYS_PER_REQUEST = 10;
+const REQUESTS_PER_BATCH = 5;
+const MAX_PUBKEYS_PER_POLL = 5 * PUBKEYS_PER_REQUEST * REQUESTS_PER_BATCH;
+
+/**
+ * Divide pubkeys into batches, each batch contains at most 5 http requests,
+ * each request can work on at most 40 pubkeys.
+ * @param pubkeysHex
+ */
+export function pubkeysToBatches(
+  pubkeysHex: string[],
+  maxPubkeysPerRequest: number = PUBKEYS_PER_REQUEST,
+  maxRequesPerBatch: number = REQUESTS_PER_BATCH
+): Batch[] {
+  if (!pubkeysHex || pubkeysHex.length === 0) {
+    return [[[]]];
+  }
+  const batches: Batch[] = [];
+
+  const pubkeysPerBatch = maxPubkeysPerRequest * maxRequesPerBatch;
+  let batch: Batch = [];
+  let pubkeysPerRequest: string[];
```

Comment:
Can find a similar approach without mutating scoped variables? You could have two for loops and scope the arrays inside them only with const